src/inmanta/moduletool.py

```diff
@@ -1079,13 +1073,45 @@ def _do_update(self, output_directory: str, setup_cfg: ConfigParser) -> None:
                 + "\n"
             )
 
-    def get_pyproject(self) -> str:
-        return """[build-system]
-requires = ["setuptools", "wheel"]
-build-backend = "setuptools.build_meta"
-"""
+    def get_pyproject(self, in_folder: str, warn_on_merge: bool = False) -> str:
+        """
+        Adds this to the existing config
+
+        [build-system]
+        requires = ["setuptools", "wheel"]
+        build-backend = "setuptools.build_meta"
+        """
+        config_in = {}
+        if os.path.exists(os.path.join(in_folder, "pyproject.toml")):
+            with open(os.path.join(in_folder, "pyproject.toml"), "r") as fh:
+                loglevel = logging.WARNING if warn_on_merge else logging.INFO
+                LOGGER.log(
+                    level=loglevel,
+                    msg="pyproject.toml file already exists, merging. This will remove all comments from the file",
+                )
+                config_in = toml.load(fh)
+
+        try:
+            build_system = config_in.setdefault("build-system", {})
+            requires = build_system.setdefault("requires", [])
+            for req in ["setuptools", "wheel"]:
+                if req not in requires:
+                    requires.append(req)
```

Comment:
```suggestion
            requires.extend(["setuptools", "wheel"])
```
Same but a bit more concise?

Reply:
not if we want to prevent duplicate values in the list